pallets/payment/src/lib.rs

```diff
@@ -240,15 +240,12 @@ pub mod pallet {
 					T::Asset::unreserve(payment.asset, &from, payment.incentive_amount);
 					// unreserve the amount to the recipent
 					T::Asset::unreserve(payment.asset, &to, payment.amount);
-					// calculate fee charged for transaction
-					let fee_amount =
-						T::FeeHandler::apply_fees(&from, &to, payment.asset, payment.amount);
 					// transfer fee amount to marketplace
 					T::Asset::transfer(
 						payment.asset,
-						&from,
-						&T::FeeRecipientAccount::get(),
-						fee_amount,
+						&from,                 // fee is paid by payment creator
```

Comment:
aren't we missing the `unreserve` of the `fee_amount` so it doesn't fail if there are not enough funds?